src/Microsoft.DotNet.Build.Tasks/MakeResourcesPriFile.cs

```diff
@@ -0,0 +1,154 @@
+                MergePriFiles();
+                Cleanup();
+            }
+            catch (Exception e)
+            {
+                Log.LogErrorFromException(e, showStackTrace: true);
+                return false; // fail the task
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        private void Initialize()
+        {
+            if (!File.Exists(ConfigFilePath))
+            {
+                throw new ArgumentException($"MakePri config file: {ConfigFilePath} was not found.");
+            }
+
+            if (!File.Exists(RunnersPriFilePath))
+            {
+                throw new ArgumentException($"App runner resources.pri file: {RunnersPriFilePath} was not found.");
+            }
+
+            if (!File.Exists(MakePriExecutablePath))
+            {
+                throw new ArgumentException($"MakePri.exe was not found in path: {MakePriExecutablePath}");
+            }
+
+            _tempFolder = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName(), "makeprifiles");
+            _reswListFile = Path.Combine(_tempFolder, "reswlist.RESFILES");
+            _priListFile = Path.Combine(_tempFolder, "prilist.RESFILES");
+            _modifiedConfigFile = Path.Combine(_tempFolder, "modifiedconfigfile.xml");
+
+            if (!Directory.Exists(_tempFolder))
+            {
+                Directory.CreateDirectory(_tempFolder);
+            }
+        }
+
+        private void Cleanup()
+        {
+            // This files are only used by MakePri.exe so we can safely delete them after we are done merging the resources. 
+            try { Directory.Delete(_tempFolder, recursive: true); }
+            catch { }
+        }
+
+        private void WritePriListFile()
+        {
+            using (var streamWriter = new StreamWriter(File.Create(_priListFile)))
+            {
+                streamWriter.WriteLine(RunnersPriFilePath);
+            }
+        }
+
+        private void WriteReswListFile()
+        {
+            using (var streamWriter = new StreamWriter(File.Create(_reswListFile)))
+            {
+                foreach (ITaskItem file in ResWFiles)
+                {
+                    streamWriter.WriteLine(file.ItemSpec);
+                }
+            }
+        }
+        
+        private void UpdateConfigFile()
+        {
+            string text = File.ReadAllText(ConfigFilePath);
+            text = text.Replace("{reswfilelist}", _reswListFile);
+            text = text.Replace("{prireslist}", _priListFile);
+            File.WriteAllText(_modifiedConfigFile, text);
+        }
+
+        private void MergePriFiles()
+        {
+            string projectRoot = Path.GetDirectoryName(RunnersPriFilePath);
+            ProcessStartInfo startInfo = new ProcessStartInfo()
+            {
+                FileName = MakePriExecutablePath,
+                Arguments = $"versioned /o /pr {projectRoot} /cf {_modifiedConfigFile} /of {NewPriFilePath} /if {RunnersPriFilePath}",
```

Comment:
>Arguments = $"versioned /o /pr {projectRoot} /cf {_modifiedConfigFile} /of {NewPriFilePath} /if {RunnersPriFilePath}", [](start = 16, length = 118)

do we need to add quotations around the different parameters? this will help avoid problem if we have spaces in the file paths